### DISCUSSION TITLE
Why is HangoutBot proprietary? And not open-source.

### DISCUSSION THREAD

**Discussion by ghost** [maxplayle04/HangoutBotPublic#4] [Q&A] [answered]
We all know you are a proprietary **demon**. And you also hate Linux probably, god you even run Windows Server (noob). but why is it closed source?

**Comment by thatmaxplayle** [comment, marked as answer]
HangoutBot, like a lot of my projects - is closed source because there's no particular need for it to be open source. There's plenty of open-source bots and with the complexity of HangoutBot (and HB's older partner, MaxBot) there is a _lot_ of code to maintain. 

Open-source in a way obligates me to:

- Make sure the code works, and that there are no bugs.
- Actually care about where stuff is and make sure it setup correcty. 
- Remove any code which is specific to the Bot's account (eg the token etc. which is kept in the project)
- Support lots of people who don't know how to use Discord.NET (which is one of the most complex Discord API's) which I honestly don't have the effort for.
- I've had my code stolen from decompilation too many times, I can't be bothered to have code stolen again.
- Since the bot is only run from my server as a fun project, and not a serious business venture; I don't need the community contribution that HangoutBot otherwise.

However, the main reason is "an open-source bot" is not the direction I want to go in with HangoutBot. I'm not the only developer - and we want this to be more of a service rather than a project. It's for the community to use the bot, and not for them to wander around the inner workings of it. 

In summary, it's not open source because _I don't want to open source it._

I have other Open Source projects, which is all I plan on making open source at this point in time :)

**Reply by ghost** [reply]
I think the argument "Make sure the code works, and that there are no bugs." is directly why open-source is so beautiful, people read your code and point out issues to you with GitHub's issue feature or they contribute and fork your project. 

"Actually care about where stuff is and make sure it setup correcty." I think this would be a best practice for any project, if something is not set up correctly it won't work or cause even more bugs

"Remove any code which is specific to the Bot's account (eg the token etc. which is kept in the project)", This can be done easily with GitHub's [.gitignore file](https://www.gitignore.io) or environment variables

"Support lots of people who don't know how to use Discord.NET (which is one of the most complex Discord API's) which I honestly don't have the effort for.", That may happen but it is not lots of them I would think about 90% would know Discord.net and then 25% of that 10% would actually join your Discord and not Google it.

"I've had my code stolen from decompilation too many times, I can't be bothered to have code stolen again.", That is the exact reason licenses exist, you can add copyright so if they end-up using a feature and saying it is not from your code you can still get it taken down. Almost all licenses require you to leave a notice of them. For your project I would recommend the GPL license if you are going to open-source it one day

"Since the bot is only run from my server as a fun project, and not a serious business venture; I don't need the community contribution that HangoutBot otherwise.", Then you choose to only add contributors and not maintainers, and open-source does **not** mean starting a business you do not make money off of it most times.

"In summary, it's not open source because I don't want to open source it.", I respect that.

**Reply by iamnotharry** [reply]
I can't speak for Max in terms of majority of the points. Although, Max being lead developer may choose to override anything I say here. The idea of open-sourcing did cross our minds in the early stages, although we resisted doing so for a couple of reasons, which I won't discuss because over-all, it's completely pointless.

With regards to your point about "good practice" (with formatting and factoring a project well) - I work with Max in a professional capacity also, and we've over time developed our own internal code conventions, which we (as well as the team we work within) all know well. This would confuse outsiders, because there's very little chance they'll have been exposed to such conventions. It's "bad practice" in the bigger picture, but it works for the developers who were intended to see the code, and that's all that matters really.

I know for a fact that Max has had licensing issues before. There is a license comment at the top of _**every single**_ CS file we produce containing the license, as there is in all of Max's projects. He's still been stolen from, and authoritative figures in the appropriate communities have done next to nothing to prevent it. This heavily contributes to the reasoning why this project is closed-source.

Because this project is not open-sourced at this moment in time, I'd like to mention that this isn't to say we'll _never_ open source it, or a version of it - but for the minute, our source code is for our eyes only. That said, I can confirm that there's no plans to open-source at this moment in time. Heck, maybe if the project dies off one day?

I'd like to reiterate, we're going for more of a service here. If Microsoft open-sourced the entirety of Windows, they'd never make any money from it, would they? 

However the main part is until Max wants to open source the project, it will stay in this condition. At the end of the day, it's a cleanup/rewrite of Max's older bot, MaxBot. So; he's in control. I'm sure there'll be further word on this from him, though.
